packages/framework/src/Model/Product/ProductFactory.php

```diff
@@ -39,6 +50,25 @@ public function createMainVariant(ProductData $data, array $variants): Product
     {
         $classData = $this->entityNameResolver->resolve(Product::class);
 
-        return $classData::createMainVariant($data, $variants);
+        $mainVariant = $classData::createMainVariant($data, $variants);
+        $this->setMissingCalculatedAvailability($mainVariant);
+
+        return $mainVariant;
+    }
+
+    /**
+     * When creating new product, ProductData::$availability is set to Product::$calculatedAvailability property as well.
+     * This is a problem when ProductData::$availability === null 
```

Comment:
I would put the sentence on one line:
```suggestion
     * This is a problem when ProductData::$availability === null as $calculatedAvailability is not nullable (unlike $availability).
```